src/ContentHubClientTrait.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+namespace Acquia\ContentHubClient;
+
+use Acquia\ContentHubClient\Guzzle\Middleware\RequestResponseHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LogLevel;
+
+/**
+ * Common trait for CH Client and CH Logging Client.
+ */
+trait ContentHubClientTrait {
```

Comment:
The implementation of get, put, post, delete can go here and we can use httpClient to carry out those requests.